examples/advanced_operations/add_responsive_search_ad_full.py

```diff
@@ -0,0 +1,538 @@
+    ad_group_response = ad_group_service.mutate_ad_groups(
+        customer_id=customer_id, operations=[ad_group_operation]
+    )
+    ad_group_resource_name = ad_group_response.results[0].resource_name
+    print(f"Created ad group {ad_group_resource_name}.")
+    return ad_group_resource_name
+
+
+def create_ad_group_ad(client, customer_id, ad_group_resource_name):
+    """Creates ad group ad.
+    
+    Args:
+      client: an initialized GoogleAdsClient instance.
+      customer_id: a client customer ID.
+      ad_group_resource_name: an ad group resource name.
+    
+    Returns:
+      Ad group ad resource name.
+    """
+    ad_group_ad_service = client.get_service("AdGroupAdService")
+
+    ad_group_ad_operation = client.get_type("AdGroupAdOperation")
+    ad_group_ad = ad_group_ad_operation.create
+    ad_group_ad.status = client.enums.AdGroupAdStatusEnum.ENABLED
+    ad_group_ad.ad_group = ad_group_resource_name
+
+    # Set responsive search ad info.
+    # https://developers.google.com/google-ads/api/reference/rpc/v11/ResponsiveSearchAdInfo
+
+    # The list of possible final URLs after all cross-domain redirects for the ad.
+    ad_group_ad.ad.final_urls.append("https://www.example.com/")
+
+    # Set a pinning to always choose this asset for HEADLINE_1. Pinning is
+    # optional; if no pinning is set, then headlines and descriptions will be
+    # rotated and the ones that perform best will be used more often.
+
+    # Headline 1
+    served_asset_enum = client.enums.ServedAssetFieldTypeEnum.HEADLINE_1
+    pinned_headline = create_ad_text_asset(
+        client, "Headline 1 testing", served_asset_enum
+    )
+
+    # Headline 2 and 3
+    ad_group_ad.ad.responsive_search_ad.headlines.extend(
+        [
+            pinned_headline,
+            create_ad_text_asset(client, "Headline 2 testing"),
+            create_ad_text_asset(client, "Headline 3 testing"),
+        ]
+    )
+
+    # Description 1 and 2
+    ad_group_ad.ad.responsive_search_ad.descriptions.extend(
+        [
+            create_ad_text_asset(client, "Desc 1 testing"),
+            create_ad_text_asset(client, "Desc 2 testing"),
+        ]
+    )
+
+    # Paths
+    # First and second part of text that can be appended to the URL in the ad.
+    # If you use the examples below, the ad will show
+    # https://www.example.com/all-inclusive/deals
+    ad_group_ad.ad.responsive_search_ad.path1 = "all-inclusive"
+    ad_group_ad.ad.responsive_search_ad.path2 = "deals"
+
+    # Send a request to the server to add a responsive search ad.
+    ad_group_ad_response = ad_group_ad_service.mutate_ad_group_ads(
+        customer_id=customer_id, operations=[ad_group_ad_operation]
+    )
+
+    for result in ad_group_ad_response.results:
+        print(
+            f"Created responsive search ad with resource name "
+            f'"{result.resource_name}".'
+        )
+        ad_resource_name = result.resource_name.split("/")[-1]
+
+    return ad_resource_name
```

Comment:
Looks like this value isn't used by the calling code, so no need to even return it at all, just remove everything below the print statement.

Reply:
TY!